### PR TITLE
Feature/ptable

### PR DESCRIPTION
**Pull request by whaeck**
Final pieces of the CE table: the unresolved resonance probability tables

### CODE REVIEW

File: python/src/block/ProbabilityTableBlock.python.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  using ProbabilityTable = njoy::ACEtk::block::ProbabilityTable;
+
+  // wrap views created by this block
+
+  // create the block
+  python::class_< Block > block(
+
+    module,
+    "ProbabilityTableBlock",
+    "he continuous energy UNR block with the unresolved resonance probability tables"
```
Comment (cjosey) [inline python/src/block/ProbabilityTableBlock.python.cpp:28]: Missing `T`.

File: python/src/block/ProbabilityTableBlock.python.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  .def_property_readonly(
+
+    "IOA",
+    &Block::IOA,
+    "The absorption competition flag"
+  )
+  .def_property_readonly(
+
+    "absorption_competion",
+    &Block::absorptionCompetion,
```
Comment (cjosey) [inline python/src/block/ProbabilityTableBlock.python.cpp:108]: This variable and associated documentation seems to be missing a `t`.

File: src/ACEtk/ContinuousEnergyTable/src/generateBlocks.hpp [hunk 3]
```diff
@@ -141,6 +141,12 @@ void generateBlocks() {
   this->yp_ = block::YP( present ? iterators.first : begin,
                          present ? iterators.second : begin );
 
+  // delayed neutron data: delayed nubar for fissile isotopes
```
Comment (cjosey) [inline src/ACEtk/ContinuousEnergyTable/src/generateBlocks.hpp:144]: This comment is for the DNU block, not URR.

File: src/ACEtk/ContinuousEnergyTable/test/ContinuousEnergyTable.test.cpp [hunk 4]
```diff
@@ -413,6 +413,22 @@ void verifyChunkU235( const ContinuousEnergyTable& chunk ) {
   CHECK( 2 == chunk.YP().index( 18 ) );
   CHECK( 3 == chunk.YP().index( 102 ) );
 
+  // DNU block
```
Comment (cjosey) [inline src/ACEtk/ContinuousEnergyTable/test/ContinuousEnergyTable.test.cpp:416]: Same comment here.

File: src/ACEtk/block/ProbabilityTable/src/generateXSS.hpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  xss.reserve( 6 * probabilities.size() );
+
+  if ( ( total.size() != bins ) || ( elastic.size() != bins ) ||
+       ( fission.size() != bins ) || ( capture.size() != bins ) ||
+       ( heating.size() != bins ) ) {
+
+    Log::error( "The size of data in the probability table are not the same" );
+    Log::info( "Expected size = {}", bins );
+    Log::info( "Found probabilities ize = {}", probabilities.size() );
+    Log::info( "Found total ize = {}", total.size() );
```
Comment (cjosey) [inline src/ACEtk/block/ProbabilityTable/src/generateXSS.hpp:20]: All `size` were written as `ize` here and the next few lines.

File: src/ACEtk/block/ProbabilityTableBlock.hpp [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+  /**
+   *  @brief Return the absorption competition flag
+   */
+  int IOA() const {
+
+    return this->empty() ? 0 : static_cast< int >( this->XSS( 5 ) );
+  }
+
+  /**
+   *  @brief Return the inelastic competition flag
```
Comment (cjosey) [inline src/ACEtk/block/ProbabilityTableBlock.hpp:109]: This comment is not in sync with `IOA`.

**Review by cjosey — APPROVED** [review]
Only a few minor documentation issues.

File: python/src/block/ProbabilityTableBlock.python.cpp (hunk 2 above)
Comment (whaeck) [reply python/src/block/ProbabilityTableBlock.python.cpp:108]: How is this possible? It's not even an English word. It also happened for the inelastic competition. I updated the C++ and Python interface and updated tests for both.

File: src/ACEtk/ContinuousEnergyTable/test/ContinuousEnergyTable.test.cpp (hunk 4 above)
Comment (whaeck) [reply src/ACEtk/ContinuousEnergyTable/test/ContinuousEnergyTable.test.cpp:416]: This one is fixed in a later PR.

File: src/ACEtk/block/ProbabilityTable/src/generateXSS.hpp (hunk 5 above)
Comment (whaeck) [reply src/ACEtk/block/ProbabilityTable/src/generateXSS.hpp:20]: I've done a refactor later that removed this file from ProbabilityTable

**Pull request merged**